### PR TITLE
Markdown example: inline HTML passthrough

### PR DESCRIPTION
**Pull request by egonSchiele**
Adds inline HTML passthrough to the markdown example — open tags, close tags, self-closing tags, and HTML comments — so common snippets like `Hello <span class="hi">world</span>!` and `<br/>` mid-paragraph round-trip through the AST instead of being mangled by the literal-`<` fallback.

## Commits

| SHA | Type | Summary |
| --- | --- | --- |
| 5db2c80 | feat | `InlineHTML` AST type added to the `InlineMarkdown` union |
| 4200449 | feat | `htmlOpenTagParser` (open + self-closing) plus shared tag-name / attribute helpers |
| 0aefdb2 | feat | `htmlCloseTagParser` for `</tag>` |
| f4e1811 | feat | `htmlCommentParser` for `<!-- … -->` |
| eca4e75 | feat | Dispatch via `htmlInlineParser` in `inlineMarkdownParser` (after autolinks, before text) |
| 1f45a77 | test | End-to-end smoke test through `markdownParser` |

## Design notes

- **One opaque AST node.** Every shape collapses to a single `InlineHTML = { type: "inline-html", content: string }` carrying the raw source (including angle brackets). No tree balancing, no sanitisation. Downstream renderers pass it through.
- **Shared building blocks** in [tests/examples/markdown/inline.ts](https://github.com/egonSchiele/tarsec/blob/feature/markdown-inline-html/tests/examples/markdown/inline.ts): `htmlTagName` (letter + `[a-zA-Z0-9-]*`), `htmlAttrName`, `htmlAttrValue` (`"…"`/`'…'`/unquoted), `htmlAttribute`, `htmlAttributes`, `htmlWS`/`htmlWS1`. All combinator-built via `seqC` + named `capture` so the reconstructed string mirrors the source exactly.
- **No hand-rolled `(input) => {...}` wrappers in this PR.** The comment parser is the case the plan flagged as "hand-rolled"; I worked through it as pure combinators instead. CommonMark's three constraints (no `-->` inside body, no `--` inside body, body can't start/end with `>`) are all baked into the body atom set:
  - `not(str("-->"))` stops at the closer
  - `not(str("--"))` rejects mid-body `--`
  - `not(seqR(char(">"), str("-->")))` is the end-of-body `>` rule (a `>` directly before the closer is rejected)
  - and a single `not(char(">"))` placed before the body's `many1` handles start-of-body `>`. An empty body falls through to `optional`'s null branch, which leaves input unconsumed so the closer matches immediately. See [the `commentBody` block](https://github.com/egonSchiele/tarsec/blob/feature/markdown-inline-html/tests/examples/markdown/inline.ts) with a commented derivation.
- **Dispatch order in `htmlInlineParser`'s `or`:** comment → close → open. Comment first so `<!--…-->` isn't intercepted by an open-tag attempt at `<!`. Close before open so `</a>` wins cleanly.
- **Dispatch order in `inlineMarkdownParser`'s `or`:** `htmlInlineParser` is inserted **after** `autolinkParser`/`bareUrlAutolinkParser` so `<https://x>` still produces an `inline-link` and `<a@b.com>` still produces a `mailto:` link; the inline HTML parsers only fire when the input is actually tag- or comment-shaped.
- **`paragraphParser` and `htmlBlockParser` unchanged.** The block form still wins at line start (no overlap because the block parser runs before any inline dispatch); the inline form only competes inside an already-running inline context.
- **No CHANGELOG/README update.** Following the PR #8 feedback that we don't need to update those for example-only additions.

## Edge cases covered by tests

- `<a>`, `<span>`, `<a href>`, `<a href="x">`, `<a href='x'>`, `<a href="x" class="y">`, `<a  href = "x"  >`, `<br/>`, `<br />`
- `</a>`, `</span>`, `</a >`, `</a   >`
- `<!---->` (empty body), `<!-- hi -->`, `<!-- line1\nline2 -->`, `<!-- foo > bar -->` (middle `>` allowed)
- Rejected: `<!-- a -- b -->` (mid-body `--`), `<!-->-->` (leading `>`), `<!-- a>-->` (trailing `>`)
- `< /a>` (space-before-tag-name) — fails as a close tag
- `<https://x>` → `inline-link` (autolink wins)
- `<a@b.com>` → `inline-link` with `mailto:` (autolink wins)
- Smoke test: `Hello <span class="hi">world</span>!` round-trips to `[text, html, text, html, text]`

## Validation

- `npx vitest run` — **463 / 463 passing** (was 445 on `main`; +18 new tests across the four phases plus the smoke test)
- `npm run test:tsc` — clean modulo the two pre-existing errors on `main` (`regex` export, `vitest.globals` module)

## Files touched

- `tests/examples/markdown/types.ts` — `InlineHTML` added to the union
- `tests/examples/markdown/inline.ts` — helpers + three parsers + `htmlInlineParser` union + dispatch
- `tests/examples/markdown/inline.test.ts` — four `describe` blocks (open/close/comment/dispatch)
- `tests/examples/markdown/index.test.ts` — one end-to-end paragraph round-trip

### CODE REVIEW

File: tests/examples/markdown/inline.ts [hunk 1]
```diff
@@ -341,6 +342,172 @@ export const bareUrlAutolinkParser: Parser<InlineLink> = map(
...
+
+const htmlAttrName: Parser<string> = map(
+  seqC(
+    capture(or(letter, char("_"), char(":")), "first"),
+    capture(manyWithJoin(or(alphanum, oneOf("_:.-"))), "rest")
+  ),
+  ({ first, rest }) => first + rest
+);
+
+const dqAttrValue: Parser<string> = map(
```
Comment (egonSchiele) [inline tests/examples/markdown/inline.ts:378]: what are dq and sq? Use more descriptive names
